### PR TITLE
`Development`: Fix automatic inference of Artemis version for docker deployments

### PR DESCRIPTION
**Pull request by daniels98it**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Compiling Artemis client using Docker-compose, or calling the npm script "webapp:prod", results in an Artemis version like "DEV". (see screenshot)

### Description

This is because the `APP_VERSION` is not injected by Gradle in this case, leading to the fallback version "DEV". As the version is specified in `build.gradle`, we can directly infer it, if this hasn't been previously injected by Gradle itself.

### Steps for Testing

Prerequisites:

* Compatible Node.js version installed

1. Compile Artemis client using Docker-compose (or directly calling `npm run start-docker` or `npm run webapp:prod`)
2. Deploy it.
3. The Artemis version displayed on the top left side should be the correct one

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Screenshots

**Before:**
Screenshot 2022-02-21 175649

**After:**
Screenshot 2022-02-21 175723

### CODE REVIEW

File: webpack/environment.js [hunk 1]
```diff
@@ -1,6 +1,34 @@
 module.exports = {
     I18N_HASH: 'generated_hash',
     SERVER_API_URL: '',
-    __VERSION__: process.env.hasOwnProperty('APP_VERSION') ? process.env.APP_VERSION : 'DEV',
+    __VERSION__: process.env.hasOwnProperty('APP_VERSION') ? process.env.APP_VERSION : inferVersion(),
```
Comment (Strohgelaender) [inline webpack/environment.js:4]:
```suggestion
    __VERSION__: process.env.APP_VERSION || inferVersion(),
```

File: webpack/environment.js [hunk 2]
```diff
@@ -1,6 +1,34 @@
...
+
+        let data = fs.readFileSync(buildGradleFile, 'UTF-8');
+
+        version = data.match(/\nversion\s=\s"(.*)"/);
+
+        if (version) {
+            version = version[1];
+        } else {
+            version = 'DEV';
+        }
```
Comment (ole-ve) [inline webpack/environment.js:28]:
```suggestion
        version = version[1] ?? 'DEV';
```

File: webpack/environment.js [hunk 3]
```diff
@@ -1,6 +1,34 @@
...
     __DEBUG_INFO_ENABLED__: false,
 };
+
+/*
+ * Needed for client compilations with docker-compose, where the 'APP_VERSION' property isn't injected by gradle.
+ *
+ * Returns the inferred APP_VERSION from 'build.gradle', or 'DEV' if this couldn't be retrieved
+ */
+function inferVersion() {
+    let version = 'DEV';
```
Comment (ole-ve) [inline webpack/environment.js:14]:
```suggestion
    let version;
```

**Review by ole-ve — COMMENTED** [review]
Added some inline comments, only style suggestions.

File: webpack/environment.js (hunk 1 above)
Comment (daniels98it) [reply webpack/environment.js]: Looks good, thanks 🙂

File: webpack/environment.js (hunk 2 above)
Comment (daniels98it) [reply webpack/environment.js]: Didn't know the '??' operator until now, but seems quite useful.. 👍 
Thanks 🙂

File: webpack/environment.js (hunk 3 above)
Comment (daniels98it) [reply webpack/environment.js:14]: Would leave this one, as is..
If an error occurs within the `try-catch` section, a `null` value would be returned..

Comment (ole-ve) [reply webpack/environment.js:14]: Yes,you are right. Did not see that 👍

**Review by ole-ve — APPROVED** [review]
Code approve. 👍

File: webpack/environment.js [hunk 4]
```diff
@@ -1,6 +1,30 @@
...
+        let data = fs.readFileSync(buildGradleFile, 'UTF-8');
+
+        version = data.match(/\nversion\s=\s"(.*)"/);
+
+        version = version[1] ?? 'DEV';
+    } catch (error) {
+        console.log("Error while retrieving 'APP_VERSION' property");
+    }
+
+    return version;
```
Comment (Strohgelaender) [inline webpack/environment.js:29]: That's minor, but I don't like the fact that `DEV` is repeated twice in the code.:
```suggestion
    const dev = 'DEV';
    try {
        const fs = require('fs');
        const buildGradleFile = 'build.gradle';

        const data = fs.readFileSync(buildGradleFile, 'UTF-8');

        const version = data.match(/\nversion\s=\s"(.*)"/);

        return version[1] ?? dev;
    } catch (error) {
        console.log("Error while retrieving 'APP_VERSION' property");
    }
    return dev;
```

**Pull request merged**